viam-orb-slam3.go

```diff
@@ -269,64 +261,50 @@ func (slamSvc *builtIn) Position(ctx context.Context, name string, extra map[str
 
 	var pInFrame *referenceframe.PoseInFrame
 	var returnedExt map[string]interface{}
+	var componentReference string
 
 	// TODO: Once RSDK-1053 (https://viam.atlassian.net/browse/RSDK-1066) is complete the original code before extracting position
 	// from GetPosition will be removed and the GetPositionNew -> GetPosition
-	if slamSvc.dev {
-		slamSvc.logger.Debug("IN DEV MODE (position request)")
+	if orbSvc.dev {
+		orbSvc.logger.Debug("IN DEV MODE (position request)")
 		req := &pb.GetPositionNewRequest{Name: name}
 
-		resp, err := slamSvc.clientAlgo.GetPositionNew(ctx, req)
+		resp, err := orbSvc.clientAlgo.GetPositionNew(ctx, req)
 		if err != nil {
 			return nil, errors.Wrap(err, "error getting SLAM position")
 		}
 
-		pInFrame = referenceframe.NewPoseInFrame(resp.GetComponentReference(), spatialmath.NewPoseFromProtobuf(resp.GetPose()))
+		componentReference = resp.GetComponentReference()
+		pInFrame = referenceframe.NewPoseInFrame(componentReference, spatialmath.NewPoseFromProtobuf(resp.GetPose()))
```

Comment:
[q] For my own learning, why did you do this?

Reply:
I did this because `slamUtils.CheckQuaternionFromClientAlgo(...)` requires a componentReference.

Reply:
Sorry, I meant: Why did you define `componentReference` outside of `NewPoseInFrame`? As compared to how it was before?

Reply:
Because `slamUtils.CheckQuaternionFromClientAlgo(...)` is in the parent scope of this line and the resp only exists within the scope of this if statement.